nixos/modules/services/databases/postgresql.nix

```diff
@@ -255,17 +305,30 @@ in
         # Wait for PostgreSQL to be ready to accept connections.
         postStart =
           ''
-            while ! ${pkgs.sudo}/bin/sudo -u ${cfg.superUser} psql --port=${toString cfg.port} -d postgres -c "" 2> /dev/null; do
+            PSQL="${pkgs.sudo}/bin/sudo -u ${cfg.superUser} psql --port=${toString cfg.port}"
```

Comment:
I would move all this into a separate service that runs as `cfg.superUser` instead, because using `sudo` here uses PAM and also is dependant on the configuration of sudoers. While at it I'd probably also remove the `postStart` step altogether and use systemd notify, since PostgreSQL has support for that since quite a while.

Reply:
@aszlig For other services which depend on having a postgresql database provisioned and have `after = [ "postgresql.service" ];` in their systemd unit will this work? My thought was that systemd would spawn both the separate service you are talking about *and* the service which depends on having a postgresql database provisioned at the same time, which could lead to problems. My systemd skills are very novice, so please forgive my ignorance here... :disappointed: